cmd/zed/lake/index/add.go

```diff
@@ -0,0 +1,119 @@
+package index
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	zedlake "github.com/brimdata/zed/cmd/zed/lake"
+	"github.com/brimdata/zed/pkg/charm"
+	"github.com/brimdata/zed/pkg/rlimit"
+	"github.com/segmentio/ksuid"
+)
+
+var Add = &charm.Spec{
+	Name:  "add",
+	Usage: "add [options] [ -rule ruleid ] tag [ ...tag ]",
+	Short: "index one or more tags",
+	New:   NewAdd,
+}
+
+type AddCommand struct {
+	lake   *zedlake.Command
+	commit bool
+	rids   idArray
+	zedlake.CommitFlags
+}
+
+func NewAdd(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
+	c := &AddCommand{lake: parent.(*Command).Command}
+	f.BoolVar(&c.commit, "commit", false, "commit added index rule if successfully written")
+	f.Var(&c.rids, "rule", "id of rule(s) to index (can be specified multiple times")
+	c.CommitFlags.SetFlags(f)
+	return c, nil
+}
+
+func (c *AddCommand) Run(args []string) error {
+	ctx, cleanup, err := c.lake.Init()
+	if err != nil {
+		return err
+	}
+	defer cleanup()
+
+	if _, err := rlimit.RaiseOpenFilesLimit(); err != nil {
+		return err
+	}
+
```

Comment:
We had agreed on a coding convention to not put blank lines like this but add a comment explaining the need for the separation if warranted.

Reply:
I was not aware of this, will fix.